app/models/response.rb

```diff
@@ -271,7 +271,7 @@ def populate_from_j2me(data)
   # Groups answers by questioning.
   # Makes sure there are associated answer objects for each questioning in the form.
   def answer_sets
-    @answer_sets ||= visible_questionings.map{ |qing| answer_set_for_questioning(qing) }
+    @answer_sets ||= Hash[visible_questionings.map{ |qing| [qing, answer_set_for_questioning(qing)] }]
```

Comment:
Hmm this works fine for me. Please give it a try again. I just fixed the index issue.